x-pack/packages/kbn-elastic-assistant/impl/assistant/api/knowledge_base/api.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  API_VERSIONS,
+  CreateKnowledgeBaseRequestParams,
+  CreateKnowledgeBaseResponse,
+  DeleteKnowledgeBaseRequestParams,
+  DeleteKnowledgeBaseResponse,
+  ELASTIC_AI_ASSISTANT_KNOWLEDGE_BASE_URL,
+  ReadKnowledgeBaseRequestParams,
+  ReadKnowledgeBaseResponse,
+} from '@kbn/elastic-assistant-common';
+import { HttpSetup, IHttpFetchError } from '@kbn/core-http-browser';
+
+/**
+ * API call for getting the status of the Knowledge Base. Provide
+ * a resource to include the status of that specific resource.
+ *
+ * @param {Object} options - The options object.
+ * @param {HttpSetup} options.http - HttpSetup
+ * @param {string} [options.resource] - Resource to get the status of, otherwise status of overall KB
+ * @param {AbortSignal} [options.signal] - AbortSignal
+ *
+ * @returns {Promise<ReadKnowledgeBaseResponse | IHttpFetchError>}
+ */
+export const getKnowledgeBaseStatus = async ({
+  http,
+  resource,
+  signal,
+}: ReadKnowledgeBaseRequestParams & { http: HttpSetup; signal?: AbortSignal | undefined }): Promise<
+  ReadKnowledgeBaseResponse | IHttpFetchError
```

Comment:
Consider allowing errors to propagate as an alternative to returning `IHttpFetchError` in this function, and the others in this file

Reply:
This was one of the existing hooks that got moved, so will live as-is for the time being. The new hooks propagate errors and return default data so are easier to work with. Will update these as the the cleanup work around the KB continues.